src/utils/shadowRoot.ts

```diff
@@ -0,0 +1,10 @@
+export const getShadowRoot = (node: any) => {
+  let parent = node && node.parentNode;
+  while (parent) {
+    if (parent.toString() === "[object ShadowRoot]") {
+      return parent;
+    }
+    parent = parent.parentNode;
+  }
```

Comment:
Is there any other option except stringifying a node? Doesn't seem very reliable)

I'm not a big expert in shadow root APIs, but I guess this might work `if (parent instanceof ShadowRoot)`